voluptuous/tests/tests.py

```diff
@@ -557,3 +557,78 @@ def fn(arg):
         return "hello"
 
     assert_raises(Invalid, fn, 1)
+
+def test_number_validation_with_string():
+    """ test with Number with string"""
+    schema = Schema({"number" : Number(precision=6, scale=2)})
+    try:
+        schema({"number": 'teststr'})
+    except MultipleInvalid as e:
+        assert_equal(str(e),
+                     "Value must be a number enclosed with string for dictionary value @ data['number']")
+    else:
+        assert False, "Did not raise Invalid for String"
+
+def test_number_validation_with_invalid_precision_scale():
+    """ test with Number with invalid precision and scale"""
+    schema = Schema({"number" : Number(precision=6, scale=2)})
+    try:
+        schema({"number": '123456.712'})
+    except MultipleInvalid as e:
+        assert_equal(str(e),
+                     "Precision must be equal to 6, and Scale must be equal to 2 for dictionary value @ data['number']")
+    else:
+        assert False, "Did not raise Invalid for String"
+
+def test_number_validation_with_valid_number():
+    """ test with Number with valid precision and scale"""
+    schema = Schema({"number" : Number(precision=6, scale=2)})
+    out_ = schema({"number": '1234.00'})
+    assert_equal(float(out_.get("number")), 1234.00)
+
+def test_number_when_precision_scale_none():
+    """ test with Number with no precision and scale"""
+    schema = Schema({"number" : Number()})
+    out_ = schema({"number": '12345678901234'})
+    assert_equal(out_.get("number"), 12345678901234)
+
+def test_number_when_precision_none_n_valid_scale_1():
+    """ test with Number with no precision and valid scale case 1"""
+    schema = Schema({"number" : Number(scale=2)})
+    out_ = schema({"number": '123456789.34'})
+    assert_equal(float(out_.get("number")), 123456789.34)
+
+def test_number_when_precision_none_n_valid_scale_2():
+    """ test with Number with no precision and valid scale case 2 with zero in decimal part"""
+    schema = Schema({"number" : Number(scale=2)})
+    out_ = schema({"number": '123456789012.00'})
+    assert_equal(float(out_.get("number")), 123456789012.00)
+
+def test_number_when_precision_none_n_invalid_scale():
+    """ test with Number with no precision and invalid scale"""
```

Comment:
It's difficult to keep track of all the function names in tests since few of them are like `test_number_when_precision_none_n_invalid_scale` while others are like `test_number_when_scale_none_n_valid_precision`. Please follow a consistent convention over here, preferably add precision first and then scale.